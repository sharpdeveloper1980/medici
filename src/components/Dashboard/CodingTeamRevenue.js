import React, { Component } from 'react';

import { Chart, Line } from 'react-chartjs-2';
const Parent = (props) => {
    
     var cummulative_revenue_month=props.chartdata.cummulative_revenue_month;
     let h = new Date();
    let days = new Date(h.getFullYear(), h.getMonth() , 0).getDate();
    let days_array = [];
    for(let i=1; i<days; i++){
        days_array.push('day'+i);
        days--;
    }
    
    var draw = Chart.controllers.line.prototype.draw;
    Chart.controllers.line.prototype.draw = function() {
        draw.apply(this, arguments);
        var ctx = this.chart.chart.ctx;
        var _stroke = ctx.stroke;
        ctx.stroke = function() {
            ctx.save();
            ctx.shadowColor = 'rgba(0,0,0,0.01)';
            ctx.shadowBlur = 20;
            ctx.shadowOffsetX = 0;
            ctx.shadowOffsetY = 5;
            _stroke.apply(this, arguments);
            ctx.restore();
        };
    };

    const lineChartData = {
        labels: days_array,
        datasets: [
            {
                
                backgroundColor: 'rgba(114, 124, 245, 0.3)',
                borderColor: '#727cf5',
                data: cummulative_revenue_month,
            },
            
        ],
    };

    const lineChartOpts = {
        maintainAspectRatio: false,
        legend: {
            display: false,
        },
        tooltips: {
            intersect: false,
        },
        hover: {
            intersect: true,
        },
        plugins: {
            filler: {
                propagate: false,
            },
        },
        scales: {
            xAxes: [
                {
                    reverse: true,
                    gridLines: {
                        color: 'rgba(0,0,0,0.05)',
                    },
                    scaleLabel: {
                        display: true,
                        labelString: 'Days of month'
                      },
                },
            ],
            yAxes: [
                {
                    ticks: {
                        stepSize: 100,
                    },
                    display: true,
                    borderDash: [5, 5],
                    gridLines: {
                        color: 'rgba(0,0,0,0)',
                        fontColor: '#fff',
                    },
                    scaleLabel: {
                        display: true,
                        labelString: 'Revenue (AUD)'
                      },
                },
            ],
        },
    }; 

    return (
        <div>
                <div className="row">
                    <div className="col-sm-9">
                        <h6 style={{ fontSize:'12px' }} className="header-title mb-2 mt-1">Coding Team Cumulative Revenue</h6>
                    </div>
                  
                </div>
                <div style={{ height: '300px' }} className="mt-3 chartjs-chart">
                    <Line data={lineChartData} options={lineChartOpts} />
                </div>
               
        </div>
    );
}
class CodingTeamRevenue extends Component {

  render() {
    return (
      <div>
         <Parent chartdata={this.props.chartdata} />
      </div>        
    );
  }
}
export default CodingTeamRevenue;
