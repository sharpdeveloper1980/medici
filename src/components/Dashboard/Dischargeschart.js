import React, { useState,useEffect,Component } from 'react';
import { Table, Progress } from 'reactstrap';
const Parent = (props) => {
    
    const [done, setDone] = useState(0);
    const [progress, setProgress] = useState(0);
    const [remaining, setRemaining] = useState(0);
    var completion_pie=props.chartdata.completion_pie
    
    useEffect(()=>{
        if(completion_pie){
           
        setDone(completion_pie.done);
        setProgress(completion_pie.progress);
        setRemaining(completion_pie.remaining);
        }
    },
    [completion_pie])
    
    return (
            <div>
                <div className="row mb-4">
                    <div className="col-sm-9">
                        <h6 style={{ fontSize:'12px' }} className="header-title mb-2 mt-1">DISCHARGES</h6>
                    </div>
                  
                </div>
                <div className="row mt-4 mb-4">
                  
                    <Table responsive className="table-sm table-centered mb-0 font-14">
                        <thead className="thead-light">
                            <tr>
                                <th>Discharge type</th>
                                <th>Count</th>
                                <th style={{ width: '40%' }}>Graph</th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>Not Commenced</td>
                                <td>{ remaining }</td>
                                <td>
                                    <Progress value={remaining} style={{ height: '3px' }} />
                                </td>
                            </tr>
                            <tr>
                                <td>In Process</td>
                                <td>{ progress }</td>
                                <td>
                                    <Progress value={progress} style={{ height: '3px' }} color="info" />
                                </td>
                            </tr>
                            <tr>
                                <td>Completed</td>
                                <td>{ done }</td>
                                <td>
                                    <Progress value={done} style={{ height: '3px' }} color="warning" />
                                </td>
                            </tr>
                           
                        </tbody>
                    </Table>
                </div>
                
               
            </div>
    );
}
class Dischargeschart extends Component {

  render() {
    return (
      <div>
         <Parent chartdata={this.props.chartdata} />
      </div>        
    );
  }
}
export default Dischargeschart;