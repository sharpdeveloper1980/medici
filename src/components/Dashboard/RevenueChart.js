import React, { Component } from 'react';
import { Bar,defaults as ChartjsDefaults, Chart } from 'react-chartjs-2';
const Parent = (props) => {
    var projected=[];
    var actual=[];
    var weeklygoal=props.chartdata.weeklygoal;
    if(weeklygoal)
    {
        for(var wl=0;wl<weeklygoal.length;wl++)
        {
            projected.push(weeklygoal[wl].projected);   
            actual.push(weeklygoal[wl].actual);   
        }
      
    }
    ChartjsDefaults.global.defaultFontColor = '#8391a2';
    ChartjsDefaults.scale.gridLines.color = '#8391a2';
    ChartjsDefaults.global.defaultFontFamily =
        '-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif';
    var draw3 = Chart.controllers.bar.prototype.draw;
    Chart.controllers.bar = Chart.controllers.bar.extend({
        draw: function() {
            draw3.apply(this, arguments);
            var ctx = this.chart.chart.ctx;
            var _fill = ctx.fill;
            ctx.fill = function() {
                ctx.save();
                ctx.shadowColor = 'rgba(0,0,0,0.01)';
                ctx.shadowBlur = 20;
                ctx.shadowOffsetX = 4;
                ctx.shadowOffsetY = 5;
                _fill.apply(this, arguments);
                ctx.restore();
            };
        },
    });

    const barChartData = canvas => {
        const ctx = canvas.getContext('2d');
        var gradientStroke = ctx.createLinearGradient(0, 500, 0, 150);
        gradientStroke.addColorStop(0, '#fa5c7c');
        gradientStroke.addColorStop(1, '#727cf5');

        return {
            labels: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'],
            datasets: [
                {
                    label: 'Projected',
                    backgroundColor:'#7380f5',
                    borderColor: '#98a3b0',
                    hoverBackgroundColor: '#7380f5',
                    hoverBorderColor: '#98a3b0',
                    data: projected,
                    barPercentage: 0.7,
                    categoryPercentage: 0.5,
                },
                {
                    label: 'Actual',
                    backgroundColor: '#98a3b0', 
                    borderColor: '#98a3b0',
                    hoverBackgroundColor: '#98a3b0',
                    hoverBorderColor: '#98a3b0',
                    data: actual,
                    barPercentage: 0.7,
                    categoryPercentage: 0.5,
                },
            ],
        };
    };

    // options
    const barChartOpts = {
        maintainAspectRatio: false,
        legend: {
            display: true,
        },
        
        tooltips: {
            backgroundColor: '#1265a1',
            titleFontColor: '#fff',
            bodyFontColor: '#fff',
            bodyFontSize: 14,
            displayColors: false,
        },
        scales: {
            yAxes: [
                {
                    gridLines: {
                        display: false,
                        color: 'rgba(0,0,0,0.05)',
                    },
                    stacked: false,
                    ticks: {
                        stepSize: 2,
                    },
			scaleLabel: {
				display: true,
				labelString:'Revenues (in 100K AUD)',
			
			}
                },

            ],
            xAxes: [
                {
                    stacked: false,
                    gridLines: {
                        color: 'rgba(0,0,0,0.05)',
                    },
                },
            ],
        },
    };

    return (
        <div>
           <div className="row">
                <div className="col-sm-9">
                    <h6 style={{ fontSize:'12px' }} className="header-title mb-2 mt-1">Team Revenue(Projections Vs Actual)</h6>
                </div>
                
            </div>
            <div style={{ height: '310px' }} className="chartjs-chart">
                <Bar data={barChartData} options={barChartOpts} />
            </div>
        </div>        
      );
}
class RevenueChart extends Component {

  render() {
    return (
      <div>
         <Parent chartdata={this.props.chartdata} />
      </div>        
    );
  }
}
export default RevenueChart;
