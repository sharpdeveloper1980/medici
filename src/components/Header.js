import React, { Component,useEffect,useState } from 'react';
import { Link,useHistory } from 'react-router-dom'; 
import { useAuth0 } from "@auth0/auth0-react";
import  logo from '../assets/images/medici_logo.svg';
// import UserIcon from '../assets/images/gr.png';
import { Dropdown, DropdownMenu, DropdownToggle } from 'reactstrap';
import SimpleBar from 'simplebar-react';
import {Chardin} from 'chardin.ts';
import 'chardin.ts/chardinjs.scss';
import { Divider } from '@material-ui/core';
import { DriveEta } from '@material-ui/icons';
let chardin;
const HeaderParent = (props) => { 
  const [dropdownOpen, setDropdownOpen] = useState(false);
  const [dropdownOpen1, setDropdownOpen1] = useState(false);
  const [dropdownOpen2, setDropdownOpen2] = useState(false);
  const [title, setTitle] = useState([]);
  const history = useHistory();
  const {
    logout,
    isAuthenticated,
  } = useAuth0();
  var logoElement = document.getElementById("appLogo");
  if(logoElement)
  {
      if(!localStorage.getItem('appOpen')) {
        logoElement.click();
        localStorage.setItem('appOpen','1');
      }
  }
  const logoutApp =() => {
    localStorage.removeItem('openReportId');
    const logoutWithRedirect = () =>
    logout({
      returnTo: window.location.origin,
    });
    logoutWithRedirect();
  }
  if(!isAuthenticated)
  { 
        history.push("/"); 
  }
    let location = props.actiontoken;
    useEffect(()=>{
      window.onbeforeunload = function() {
        // logoutApp(); 
      };
      let activeusername = localStorage.getItem('activeusername');
      let name = activeusername.split(' ');
      let firstchar ='';
      let secondchar = '';
      // eslint-disable-next-line
      if(name[0].length>0){
          firstchar = name[0].charAt(0);
      }
      // eslint-disable-next-line
      if(name[1].length >0){
          secondchar = name[1].charAt(0);
      }
      setTitle(firstchar+''+secondchar);
      chardin = new Chardin(document.querySelector('body'));
      
        // $('body').on('click', 'button[data-toggle="chardinjs"]', function (e) {
        //     e.preventDefault();
        //     return ($('body').data('chardinJs')).toggle();
        // });  
    },[]);
    const showHelp=()=>{
        chardin.start();
    }
    let coderUrl='/discharges/discharge';
	  const toggleDropdown=() => {
      setDropdownOpen(!dropdownOpen);
    }
    const toggleDropdown1=() => {
      setDropdownOpen1(!dropdownOpen1);
    }
    const toggleDropdown2=() => {
      setDropdownOpen2(!dropdownOpen2);
    }
    const notificationContainerStyle = {
        maxHeight: '230px',
        display: 'none',
    };
    const returnTo=(param)=>{
        
        if(localStorage.getItem('ischanged') == 'true'){
            let ele = document.getElementById("saveonTabChange");
                if(ele)
                {
                    ele.click();
                   
                }
        }else{
            
            history.push(param);
        }
    }
    return (
      <React.Fragment>
            
      <div className="navbar-custom " style={{'height':'92px'}}>
          <div className="topnav-navbar-dark">
          <div className="container-fluid">
              <ul className="list-unstyled topbar-right-menu float-right mb-0">
                  <li className="notification-list topbar-dropdown d-lg-none">
                    <Dropdown isOpen={dropdownOpen} toggle={toggleDropdown}>
                        <DropdownToggle
                            data-toggle="dropdown"
                            tag="a"
                            className="nav-link dropdown-toggle arrow-none mr-0"
                            onClick={toggleDropdown}
                            aria-expanded={dropdownOpen}>
                            <i className="mdi mdi-magnify noti-icon"></i>
                        </DropdownToggle>
                        <DropdownMenu right className="dropdown-menu-animated topbar-dropdown-menu dropdown-lg">
                            <form className="p-3">
                                <input type="text" className="form-control" placeholder="Search ..." />
                            </form>
                        </DropdownMenu>
                    </Dropdown>
                  </li>
                  <li className="notification-list">
                    <Dropdown>
                      <DropdownToggle
                          data-toggle="dropdown"
                          tag="button"
                          className="nav-link dropdown-toggle arrow-none btn btn-link"
                         >
                          <i onClick={showHelp} className="dripicons-question noti-icon" data-toggle="chardinjs"></i>
                      </DropdownToggle>
                  </Dropdown>
                  </li>
                  <li className="notification-list" style={{'marginTop':'4px'}} >
                    <Dropdown isOpen={dropdownOpen1} toggle={toggleDropdown1}>
                      <DropdownToggle
                          data-toggle="dropdown"
                          tag="button"
                          className="nav-link dropdown-toggle arrow-none btn btn-link"
                          onClick={toggleDropdown1}
                          aria-expanded={dropdownOpen1}>
                          <i className="mdi mdi-bell-outline noti-icon"></i>
                          <span className="noti-icon-badge"></span>
                      </DropdownToggle>
                      <DropdownMenu right className="dropdown-menu-animated dropdown-lg">
                          <div onClick={toggleDropdown1}>
                              <div className="dropdown-item noti-title">
                                  <h5 className="m-0">
                                      <span className="float-right">
                                          
                                              <small>Clear All</small>
                                          
                                      </span>
                                      Notification
                                  </h5>
                              </div>
                              <SimpleBar style={{notificationContainerStyle}}>
                                          <Link style={{'display':'none'}}
                                              to="/"
                                              className="dropdown-item notify-item"
                                              key="1">
                                              <div className={`notify-icon bg-success`}>
                                                  <i className=""></i>
                                              </div>
                                              <p className="notify-details">
                                                  sdasd
                                                  <small className="text-muted">B</small>
                                              </p>
                                          </Link>
                              </SimpleBar>

                              <Link to="/" className="dropdown-item text-center text-primary notify-item notify-all">
                                  View All
                              </Link>
                          </div>
                      </DropdownMenu>
                  </Dropdown>
                  </li>
                  <li className="notification-list">
                    <Dropdown>
                      <DropdownToggle
                          data-toggle="dropdown"
                          tag="button"
                          className="nav-link dropdown-toggle arrow-none btn btn-link"
                         >
                          <i className="dripicons-gear noti-icon noti-icon"></i>
                      </DropdownToggle>
                  </Dropdown>
                  </li>
                  <li className="notification-list">
                    <Dropdown isOpen={dropdownOpen2} className="mt-1" toggle={toggleDropdown2}>
                      <DropdownToggle
                          data-toggle="dropdown"
                          tag="button"
                          className="nav-link dropdown-toggle nav-user arrow-none mr-0"
                          onClick={toggleDropdown2}
                          aria-expanded={dropdownOpen2}>
                          <span className=" account-user-avatar">
                               {/* <img src={localStorage.getItem('activeuserimage')} className="rounded-circle" alt="user" />  */}
                              <div className="rounded-circle" alt="user">
                                    <span class="avatar-title bg-success rounded-circle" style={{padding:'0.4rem'}}>
                                                { title }
                                    </span>
                              </div>
                          </span>
                          
                          <span>
                              <span className="account-user-name" style={{marginLeft:'0.5rem'}}>{localStorage.getItem('activeusername')}</span>
                              <span className="account-position" style={{marginLeft:'0.5rem'}}>Coder</span>
                          </span>
                      </DropdownToggle>
                      <DropdownMenu right className="dropdown-menu-animated topbar-dropdown-menu profile-dropdown">
                          <div onClick={toggleDropdown2}>
                              <div className="dropdown-header noti-title">
                                  <h6 className="text-overflow m-0">Welcome !</h6>
                              </div>
                            <div
                                onClick={()=>returnTo('/dashboard')}
                                className="dropdown-item notify-item"
                                key="1">
                                <i className="mdi mdi-account-circle mr-1"></i>
                                <span>My Account</span>
                            </div>
                            <div
                                onClick={()=>returnTo('/dashboard')}
                                className="dropdown-item notify-item"
                                key="1">
                                <i className="mdi mdi-account-edit mr-1"></i>
                                <span>Setting</span>
                            </div>
                            <div
                                onClick={()=>returnTo('/dashboard')}
                                className="dropdown-item notify-item"
                                key="1">
                                <i className="mdi mdi-lifebuoy mr-1"></i>
                                <span>Support</span>
                            </div>
                            <div
                                onClick={()=>returnTo('/dashboard')}
                                className="dropdown-item notify-item"
                                key="1">
                                <i className="mdi mdi-lock-outline mr-1"></i>
                                <span>Lock Screen</span>
                            </div>
                            <span style={{'cursor':'pointer'}}
                                onClick={logoutApp}
                                className="dropdown-item notify-item"
                                key="1">
                                <i className="mdi mdi-logout mr-1"></i>
                                <span>Logout</span>
                            </span>
                          </div>
                      </DropdownMenu>
                  </Dropdown>
                  </li>
              </ul>
              <div returnTo={()=>returnTo('/dashboard')} id="appLogo"><img alt="MediciNos" src={logo} style={{'width':'100px','height':'65px'}} /></div>
          </div>
          </div>
          <div className="card second_header shadow-sm">
            <div className="card-body second_header_body">
                <div class="container-fluid">
                        <nav class="navbar navbar-light navbar-expand-lg topnav-menu">
                            <div class="collapse navbar-collapse" id="topnav-menu-content">
                                <ul class="navbar-nav">
                                    <li class="nav-item" data-intro="An awesome." data-position="right">
                                        <div className={`${location === 'dashboard' ? 'nav-link showActive' : 'nav-link'}`} onClick={()=>returnTo('/dashboard')}   id="topnav-dashboards" role="button"
                                            data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                                            <i class="uil-dashboard mr-1"></i>
                                            Dashboard
                                        </div>
                                    </li>
                                    <li class="nav-item dropdown">
                                        <div className={`${location === 'discharge' ? 'nav-link showActive' : 'nav-link'}`} onClick={()=>returnTo(coderUrl)}  id="topnav-apps"
                                            role="button" data-toggle="dropdown" aria-haspopup="true"
                                            aria-expanded="false">
                                            <i class="uil-apps mr-1"></i>
                                            Discharges
                                        </div>
                                    </li>
                                </ul>
                            </div>
                        </nav>
                    </div>
            </div>
        </div>
      </div>
      
  </React.Fragment>     
      );
} 
class Header extends Component {
  render() {
    return (
      <div>
         <HeaderParent actiontoken={this.props.actiontoken}  />
      </div>        
    );
  }
}
export default Header;