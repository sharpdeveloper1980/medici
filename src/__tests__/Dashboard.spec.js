import React,{ useState,useEffect } from "react";
import { create } from "react-test-renderer";
import {  Row, Col, Card, CardBody } from 'reactstrap';
import { useAuth0 } from "@auth0/auth0-react"; 
import config from "../auth_config.json";
import Header from '../components/Header';
import RevenueChart from '../components/Dashboard/RevenueChart';
import Dischargeschart from '../components/Dashboard/Dischargeschart';
import CodingTeamRevenue from '../components/Dashboard/CodingTeamRevenue';
const Dashboard=(props)=>{
    const [dashboard_data, setData] = useState([]);
    const { getAccessTokenSilently } = useAuth0();
    useEffect(()=>{
        let responseJson={
            "Discharges": {
              "value": 364,
              "change": 3
            },
            "revenue": {
              "value": "32 M",
              "change": 2
            },
            "individual_discharges": {
              "value": 21,
              "change": 3
            },
            "individual_revenue": {
              "value": "37K",
              "change": 2
            },
            "cummulative_revenue_month": [
              0,
              120,
              180,
              244,
              354,
              440,
              520,
              600,
              680,
              720,
              770
            ],
            "completion_pie": {
              "done": 33,
              "progress": 3,
              "remaining": 43
            },
            "weeklygoal": [
              {
                "projected": 6,
                "actual": 8
              },
              {
                "projected": 6,
                "actual": 5
              },
              {
                "projected": 6,
                "actual": 5
              },
              {
                "projected": 6,
                "actual": 7
              },
              {
                "projected": 4,
                "actual": 0
              }
            ]
          };
        setData(responseJson); 
    },[]);
    /* Card date Start */
    function Discharges() {
        const Discharges = dashboard_data.Discharges;
        if (Discharges) {
            return dashboard_data.Discharges.value;
        }
        return 0;
    }
    function Revenue()
    {
        const Revenue = dashboard_data.revenue;
        if (Revenue) {
            return dashboard_data.revenue.value;
        }
        return 0;
    }
    function IndividualDischarges()
    {
        const IndividualDischarges = dashboard_data.individual_discharges;
        if (IndividualDischarges) {
            return dashboard_data.individual_discharges.value;
        }
        return 0;
    }
    function IndividualRevenue()
    {
        const IndividualRevenue = dashboard_data.individual_revenue;
        if (IndividualRevenue) {
            return dashboard_data.individual_revenue.value;
        }
        return 0;
    }
    /* Card date End */
    return (
        <div className="wrapper">
            <div className="content-page">
            <div className="content page_container">   
                    <div className="container-fluid content_box">
                        
                        <Row>
                            <Col>
                                <div className="page-title-box">
                                    <div className="page-title-right">
                                        
                                    </div>
                                    <h4 className="page-title">Dashboard</h4>
                                    <p>Welcome back, James</p>
                                </div>
                            </Col>
                        </Row>

                        <Row>
                            <Col lg={6}>
                                <div className="row">
                                    <div className="col-lg-6">
                                        <div className="card widget-flat">
                                            <div className="card-body">
                                                <div className="float-right">
                                                    <i className="mdi mdi-file-document-box-check-outline"></i>
                                                </div>
                                                <h5 className="text-muted font-weight-normal mt-0" title="Number of Discharges">Discharges</h5>
                                                <h3 className="mt-3 mb-3"><Discharges /></h3>
                                                <div className="mb-0 text-muted">
                                                    <div className="text-success mr-2"><i className="mdi mdi-arrow-up-bold"></i> 5.27%</div>
                                                    <div className="text-nowrap">Since last month</div>  
                                                </div>
                                            </div>
                                        </div>
                                    </div>

                                    <div className="col-lg-6">
                                        <div className="card widget-flat">
                                            <div className="card-body">
                                                <div className="float-right">
                                                    <i className="mdi mdi-currency-usd"></i>
                                                </div>
                                                <h5 className="text-muted font-weight-normal mt-0" title="Revenue">Revenue</h5>
                                                <h3 className="mt-3 mb-3">$ <Revenue /></h3>
                                                <div className="mb-0 text-muted">
                                                    <div className="text-danger mr-2"><i className="mdi mdi-arrow-down-bold"></i> 1.08%</div>
                                                    <div className="text-nowrap">Since last month</div>
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                                <div className="row">
                                    <div className="col-lg-6">
                                        <div className="card widget-flat">
                                            <div className="card-body">
                                                <div className="float-right">
                                                    <i className="mdi mdi-file-document-box-check"></i>
                                                </div>
                                                <h5 className="text-muted font-weight-normal mt-0" title="Number of Customers">Team Discharges </h5>
                                                <h3 className="mt-3 mb-3"><IndividualDischarges /></h3>
                                                <div className="mb-0 text-muted">
                                                    <div className="text-success mr-2"><i className="mdi mdi-arrow-up-bold"></i> 5.27%</div>
                                                    <div className="text-nowrap">Since last month</div>  
                                                </div>
                                            </div>
                                        </div>
                                    </div>

                                    <div className="col-lg-6">
                                        <div className="card widget-flat">
                                            <div className="card-body">
                                                <div className="float-right">
                                                    <i className="mdi mdi-coin-outline"></i>
                                                </div>
                                                <h5 className="text-muted font-weight-normal mt-0" title="Number of Orders">Team Revenue</h5>
                                                <h3 className="mt-3 mb-3">$ <IndividualRevenue /></h3>
                                                <div className="mb-0 text-muted">
                                                    <div className="text-danger mr-2"><i className="mdi mdi-arrow-down-bold"></i> 1.08%</div>
                                                    <div className="text-nowrap">Since last month</div>
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                            </Col>

                            <Col lg={6}>
                                <Card>
                                    <CardBody>
                                        
                                    </CardBody>
                                </Card>
                            </Col>
                        </Row>

                        <Row>
                            <Col lg={6}>
                                <Card>
                                    <CardBody>
                                       
                                    </CardBody>
                                </Card>
                            </Col>
                            <Col lg={6}>
                                <Card>
                                    <CardBody>
                                        
                                    </CardBody>
                                </Card>
                            </Col>
                            
                        </Row>
                    </div>
                </div>
            </div>


        </div>
    );
};
describe("Dashboard component", () => {
  test("Matches the snapshot", () => {
    const dashboard = create(<Dashboard />);
    expect(dashboard.toJSON()).toMatchSnapshot();
  });
});