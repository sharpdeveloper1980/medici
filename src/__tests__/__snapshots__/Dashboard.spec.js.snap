// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Matches the snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="content-page"
  >
    <div
      className="content page_container"
    >
      <div
        className="container-fluid content_box"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <div
              className="page-title-box"
            >
              <div
                className="page-title-right"
              />
              <h4
                className="page-title"
              >
                Dashboard
              </h4>
              <p>
                Welcome back, James
              </p>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-6"
          >
            <div
              className="row"
            >
              <div
                className="col-lg-6"
              >
                <div
                  className="card widget-flat"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="float-right"
                    >
                      <i
                        className="mdi mdi-file-document-box-check-outline"
                      />
                    </div>
                    <h5
                      className="text-muted font-weight-normal mt-0"
                      title="Number of Discharges"
                    >
                      Discharges
                    </h5>
                    <h3
                      className="mt-3 mb-3"
                    >
                      0
                    </h3>
                    <div
                      className="mb-0 text-muted"
                    >
                      <div
                        className="text-success mr-2"
                      >
                        <i
                          className="mdi mdi-arrow-up-bold"
                        />
                         5.27%
                      </div>
                      <div
                        className="text-nowrap"
                      >
                        Since last month
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col-lg-6"
              >
                <div
                  className="card widget-flat"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="float-right"
                    >
                      <i
                        className="mdi mdi-currency-usd"
                      />
                    </div>
                    <h5
                      className="text-muted font-weight-normal mt-0"
                      title="Revenue"
                    >
                      Revenue
                    </h5>
                    <h3
                      className="mt-3 mb-3"
                    >
                      $ 
                      0
                    </h3>
                    <div
                      className="mb-0 text-muted"
                    >
                      <div
                        className="text-danger mr-2"
                      >
                        <i
                          className="mdi mdi-arrow-down-bold"
                        />
                         1.08%
                      </div>
                      <div
                        className="text-nowrap"
                      >
                        Since last month
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-lg-6"
              >
                <div
                  className="card widget-flat"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="float-right"
                    >
                      <i
                        className="mdi mdi-file-document-box-check"
                      />
                    </div>
                    <h5
                      className="text-muted font-weight-normal mt-0"
                      title="Number of Customers"
                    >
                      Team Discharges 
                    </h5>
                    <h3
                      className="mt-3 mb-3"
                    >
                      0
                    </h3>
                    <div
                      className="mb-0 text-muted"
                    >
                      <div
                        className="text-success mr-2"
                      >
                        <i
                          className="mdi mdi-arrow-up-bold"
                        />
                         5.27%
                      </div>
                      <div
                        className="text-nowrap"
                      >
                        Since last month
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col-lg-6"
              >
                <div
                  className="card widget-flat"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="float-right"
                    >
                      <i
                        className="mdi mdi-coin-outline"
                      />
                    </div>
                    <h5
                      className="text-muted font-weight-normal mt-0"
                      title="Number of Orders"
                    >
                      Team Revenue
                    </h5>
                    <h3
                      className="mt-3 mb-3"
                    >
                      $ 
                      0
                    </h3>
                    <div
                      className="mb-0 text-muted"
                    >
                      <div
                        className="text-danger mr-2"
                      >
                        <i
                          className="mdi mdi-arrow-down-bold"
                        />
                         1.08%
                      </div>
                      <div
                        className="text-nowrap"
                      >
                        Since last month
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-lg-6"
          >
            <div
              className="card"
            >
              <div
                className="card-body"
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-6"
          >
            <div
              className="card"
            >
              <div
                className="card-body"
              />
            </div>
          </div>
          <div
            className="col-lg-6"
          >
            <div
              className="card"
            >
              <div
                className="card-body"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
